grails-app/controllers/org/pih/warehouse/inventory/StockMovementController.groovy

```diff
@@ -160,7 +165,12 @@ class StockMovementController {
 
     def removeStockMovement = {
         StockMovement stockMovement = stockMovementService.getStockMovement(params.id)
-        if (stockMovement?.shipment?.currentStatus == ShipmentStatusCode.PENDING || !stockMovement?.shipment?.currentStatus) {
+        if (stockMovement?.requisition?.origin?.id != session.warehouse.id) {
+            if (params.show) {
+                redirect(action: "show", id: params.id)
+            }
+            flash.error = "You are not able to delete stock movement from a location other than origin."
```

Comment:
The flash.error should be set above line 169 since the return I've asked to be added after the redirect would cause flash.error not to be set.

Reply:
Also, I can't click on line 193, but please wrap that line with parentheses. It took me a minute to be able to read that properly.